app/controllers/content_items_controller.rb

```diff
@@ -65,23 +65,43 @@ def redirect_canonical(content_item)
   end
 
   def can_view(item)
-    if auth_bypass_id_header.present?
-      item.includes_auth_bypass_id?(auth_bypass_id_header)
+    if valid_user?(auth_user_id)
+      if item.access_limited?
+        item.viewable_by_bypass_id?(auth_bypass_id) || item.viewable_by_user_id?(auth_user_id) || item.viewable_by_org_id?(auth_org_id)
+      else
+        true
+      end
     else
-      !invalid_user_id? && item.viewable_by?(authenticated_user_uid)
+      false
     end
   end
 
-  def authenticated_user_uid
+  def valid_user?(user_id)
+    user_id != 'invalid'
+  end
```

Comment:
I think, but my understanding could be wrong, that a user would have an user_id of "invalid" if they are using an auth bypass token: https://github.com/alphagov/authenticating-proxy/blob/84b7b5d300fe7de8b328be964c421870a412bfd0/lib/proxy.rb#L95-L100

Reply:
Yup - so having another look. Are we right in assume we don't need to do any check for `invalid` in the content store?